components/ledger/pkg/ledger/cache/cache.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"context"
+	"strings"
+
+	"github.com/bluele/gcache"
+	"github.com/formancehq/ledger/pkg/core"
+	"github.com/formancehq/ledger/pkg/storage"
+)
+
+type Cache struct {
+	cache gcache.Cache
+	store storage.LedgerStore
+}
+
+func (c *Cache) GetAccountWithVolumes(ctx context.Context, address string) (*core.AccountWithVolumes, error) {
+
+	address = strings.TrimPrefix(address, "@")
+
+	rawAccount, err := c.cache.Get(c.accountKey(address))
+	if err != nil {
+		// TODO: Rename later ?
+		account, err := c.store.ComputeAccount(ctx, address)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := c.cache.Set(c.accountKey(account.Address), account); err != nil {
+			panic(err)
+		}
+
+		return account, nil
```

Comment:
should we copy also here before returning the cache ?

Reply:
Good catch!